analysis/CompareNECRvsLength.py

```diff
@@ -0,0 +1,136 @@
+            dataList = [tracerData]
+        necr, true, rPlusS, scatters, randoms = DetectedCoincidences( activityList, dataList, simulationWindow, coincidenceWindow, detectorRadius, ZMin=-zWindow, ZMax=zWindow )
+        necrAtTime.append( necr )
+        trueAtTime.append( true )
+        rPlusSAtTime.append( rPlusS )
+        activityAtTime.append( activity / phantomVolume )
+
+    mpl.clf()
+    return activityAtTime, necrAtTime, trueAtTime, rPlusSAtTime
+
+def GetDataset(dirName, detectorLength, phantomLength, NDecays, detectorMaterial, energyMin=435.0, energyMax=585.0):
+
+    random.seed( detectorLength )
+
+    # Get tracer data
+    tracerFileName = dirName + "hits.n" + str(NDecays) + ".SiemensBlock." + str(detectorLength) + "mm.LinearF18." + str(phantomLength) + "mm.1234.csv"
+    tracerData = SimulationDataset(tracerFileName, NDecays, 435.0, 585.0)
+
+    crystalActivity = None
+    crystalData = None
+
+    if detectorMaterial == "LSO" or detectorMaterial == "LYSO" :
+        crystalActivity = sqp.Lu176decaysInMass( sqp.DetectorMassLength( detectorLength, detectorMaterial ) )
+        # Get crystal data
+        crystalFileName = dirName + "hits.n" + str(NDecays) + ".SiemensBlock." + str(detectorLength) + "mm.Siemens." + str(phantomLength) + "mm.1234.csv"
+        crystalData = SimulationDataset(crystalFileName, NDecays, 435.0, 585.0)
+    
+    activityAtTimeSiemens, necrAtTimeSiemens, trueAtTimeSiemens, rPlusSAtTimeSiemens = NECRatTimeF18( tracerData, crystalData, crystalActivity, sqp.DetectorRadius(), phantomLength, detectorMaterial, simulationWindow, coincidenceWindow )
+    return ( max( necrAtTimeSiemens ), sqp.DetectorDiscreteLength( detectorLength ) )
+
+def PeakNECRWithLengthMultiprocess(dirName, phantomLength, detectorMaterial, simulationWindow=1E-3, coincidenceWindow=4.7E-9, processes=5 ):
+    # Create the arguments for each process
+    arguments = []
+    for detectorLength in detectorLengths:
+        arguments.append( (dirName, detectorLength, phantomLength, NDecays, detectorMaterial, simulationWindow, coincidenceWindow ) )
+
+    # Launch a separate process for each detector length
+    result = None
+    with Pool( processes=processes ) as p:
+        result = p.starmap( GetDataset, arguments )
+
+    # Unpack the results
+    realLengths = []
+    maxNECR = []
+    for entry in result:
+        maxNECR.append( entry[0] )
+        realLengths.append( entry[1] )
+
+    # Useful when multiprocessing does not work, do not remove
+    # for detectorLength in detectorLengths:
+    #     result = GetDataset(dirName, detectorLength, phantomLength, NDecays, detectorMaterial, simulationWindow, coincidenceWindow)
+    #     maxNECR.append(result[0])
+    #     realLengths.append(result[1])
+    
+    return realLengths, maxNECR
+
+mpl.figure()
+legItems = ["LSO", "NaI", "BGO"]
+items = ["LSO", "NaI", "BGO"]
+dirNames = [
+    "/Users/hannaborecka-bielska/Desktop/PET/pet-sim/sim/Data/LSO-1M/",
```

Comment:
Avoid hard-coded paths

Reply:
I know this should be avoided but in this file it is required that a path is provided. I therefore would like to leave it as is. The whole file is moved now to another directory.